addons/binding/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/profiles/EnOceanProfileFactory.java

```diff
@@ -0,0 +1,116 @@
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.CoreItemFactory;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.DefaultSystemChannelTypeProvider;
+import org.eclipse.smarthome.core.thing.profiles.Profile;
+import org.eclipse.smarthome.core.thing.profiles.ProfileAdvisor;
+import org.eclipse.smarthome.core.thing.profiles.ProfileCallback;
+import org.eclipse.smarthome.core.thing.profiles.ProfileContext;
+import org.eclipse.smarthome.core.thing.profiles.ProfileFactory;
+import org.eclipse.smarthome.core.thing.profiles.ProfileType;
+import org.eclipse.smarthome.core.thing.profiles.ProfileTypeProvider;
+import org.eclipse.smarthome.core.thing.profiles.ProfileTypeUID;
+import org.eclipse.smarthome.core.thing.type.ChannelType;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeRegistry;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ *
+ * @author Daniel Weber - Initial contribution
+ */
+@Component(service = { ProfileFactory.class, ProfileTypeProvider.class, ProfileAdvisor.class })
+
+public class EnOceanProfileFactory implements ProfileFactory, ProfileAdvisor, ProfileTypeProvider {
+
+    private ChannelTypeRegistry channelTypeRegistry;
+
+    @Override
+    public @Nullable Profile createProfile(ProfileTypeUID profileTypeUID, ProfileCallback callback,
+            ProfileContext profileContext) {
+
+        if (profileTypeUID.equals(EnOceanProfileTypes.RockerSwitchToPlayPause)) {
+            return new RockerSwitchToPlayPauseProfile(callback);
+        } else if (profileTypeUID.equals(EnOceanProfileTypes.RockerSwitchToRollershutter)) {
+            return new RockerSwitchToRollershutterProfile(callback);
+        } else if (profileTypeUID.equals(EnOceanProfileTypes.RockerSwitchFromOnOff)) {
+            return new RockerSwitchFromOnOffProfile(callback, profileContext);
+        } else if (profileTypeUID.equals(EnOceanProfileTypes.RockerSwitchFromRollershutter)) {
+            return new RockerSwitchFromUpDownProfile(callback, profileContext);
+        }
+
+        return null;
+
+    }
+
+    @Override
+    public Collection<@NonNull ProfileTypeUID> getSupportedProfileTypeUIDs() {
+        return SUPPORTED_PROFILETYPES_UIDS;
+    }
+
+    @Nullable
+    @Override
+    public ProfileTypeUID getSuggestedProfileTypeUID(@Nullable ChannelType channelType, @Nullable String itemType) {
+
+        if (channelType == null) {
+            return null;
+        }
+
+        if (DefaultSystemChannelTypeProvider.SYSTEM_RAWROCKER.getUID().equals(channelType.getUID())) {
+            if (CoreItemFactory.PLAYER.equalsIgnoreCase(itemType)) {
+                return EnOceanProfileTypes.RockerSwitchToPlayPause;
+            } else if (CoreItemFactory.ROLLERSHUTTER.equalsIgnoreCase(itemType)) {
+                return EnOceanProfileTypes.RockerSwitchToRollershutter;
```

Comment:
Does this actually work on a `SYSTEM_RAWROCKER`? The implementation seems to expect events such as DIR1 and DIR2, so this looks rather EnOcean specific. If it is NOT EnOcean-specific, such profiles should rather be moved to ESH.